web_pwa/controllers/main.py

```diff
@@ -0,0 +1,69 @@
+from odoo.http import request, Controller, route
+
+
+class PWA(Controller):
+
+    def get_asset_urls(self, asset_xml_id):
+        qweb = request.env['ir.qweb'].sudo()
+        assets = qweb._get_asset_nodes(asset_xml_id, {}, True, True)
+        urls = []
+        for asset in assets:
+            if asset[0] == 'link':
+                urls.append(asset[1]['href'])
+            if asset[0] == 'script':
+                urls.append(asset[1]['src'])
+        return urls
+
+    @route('/service-worker.js', type='http', auth="public")
```

Comment:
```suggestion
    @route('/web_pwa/service-worker.js', type='http', auth="public")
```

Reply:
Done

Reply:
Just noticed:

> The path of the provided scope ('/') is not under the max scope allowed ('/web_pwa/'). Adjust the scope, move the Service Worker script, or use the Service-Worker-Allowed HTTP header to allow the scope.

So, service worker must be served  at `/`

Modifying

Reply:
Oh interesting detail.

Reply:
Done